src/core/Link/Link.tsx

```diff
@@ -6,17 +6,20 @@ import {
   Link as CompLink,
   LinkProps as CompLinkProps,
 } from '../../components/Link/Link';
+import { asPropType } from '../../utils/typescript';
 import { LinkExternal, LinkExternalProps } from './LinkExternal';
 import { baseStyles } from './Link.baseStyles';
 export { LinkExternal, LinkExternalProps };
 
 type LinkVariant = 'default' | 'external';
 export interface LinkProps extends CompLinkProps, ThemeComponent {
   variant?: LinkVariant;
+  asProp?: asPropType;
 }
 
-const StyledLink = styled(({ theme, ...passProps }: LinkProps) => (
-  <CompLink {...passProps} />
+const StyledLink = styled(({ theme, asProp, ...passProps }: LinkProps) => (
+  // as-property is defined internally as asProp and need to be implemented back if used
```

Comment:
I'd rephrase this "need to be implemented back if used" where ever used to better explain that asProp always needs custom implementation per component if support for it is desired.